src/dict.c

```diff
@@ -359,32 +445,52 @@ int dictAdd(dict *d, void *key, void *val)
 dictEntry *dictAddRaw(dict *d, void *key, dictEntry **existing)
 {
     long index;
-    dictEntry *entry;
-    int htidx;
-
-    if (dictIsRehashing(d)) _dictRehashStep(d);
```

Comment:
moving this has an implication that dictAddRaw that results in a NOP doesn't do incremental rehashing.
considering that dictFind does, i think it's odd.
let's consider reverting that after we decide what to do with the dictInsertAtIndex public interface.

Reply:
I've added a dictRehashStep call in dictKeyIndex to preserve this behaviour, so this isn't affected.

(I'll rename dictKeyIndex to dictFindBucketForInsert to not expose the index.)